lcls_tools/superconducting/scLinac.py

```diff
@@ -480,39 +480,48 @@ def move(
         maxSteps: int = utils.DEFAULT_STEPPER_MAX_STEPS,
         speed: int = utils.DEFAULT_STEPPER_SPEED,
         changeLimits: bool = True,
+        check_detune: bool = True,
     ):
         """
         :param numSteps: positive for increasing cavity length, negative for decreasing
         :param maxSteps: the maximum number of steps allowed at once
         :param speed: the speed of the motor in steps/second
         :param changeLimits: whether to change the speed and steps
+        :param check_detune: whether to check for valid detune after each move
         :return:
         """
 
         self.check_abort()
+        maxSteps = abs(maxSteps)
 
         if changeLimits:
             # on the off chance that someone tries to write a negative maximum
-            self.max_steps = abs(maxSteps)
+            self.max_steps = maxSteps
 
             # make sure that we don't exceed the speed limit as defined by the tuner experts
             self.speed = (
                 speed if speed < utils.MAX_STEPPER_SPEED else utils.MAX_STEPPER_SPEED
             )
 
         if abs(numSteps) <= maxSteps:
-            print(f"{self.cavity} {numSteps} steps <= {maxSteps} max")
+            print(f"{self.cavity} {abs(numSteps)} steps <= {maxSteps} max")
             self.step_des = abs(numSteps)
-            self.issueMoveCommand(numSteps)
+            self.issueMoveCommand(numSteps, check_detune=check_detune)
             self.restoreDefaults()
         else:
-            print(f"{self.cavity} {numSteps} steps > {maxSteps} max")
+            print(f"{self.cavity} {abs(numSteps)} steps > {maxSteps} max")
             self.step_des = maxSteps
-            self.issueMoveCommand(numSteps)
+            self.issueMoveCommand(numSteps, check_detune=check_detune)
             print(f"{self.cavity} moving {numSteps - (sign(numSteps) * maxSteps)}")
-            self.move(numSteps - (sign(numSteps) * maxSteps), maxSteps, speed, False)
+            self.move(
+                numSteps - (sign(numSteps) * maxSteps),
+                maxSteps,
+                speed,
+                changeLimits=False,
+                check_detune=check_detune,
+            )
 
-    def issueMoveCommand(self, numSteps):
+    def issueMoveCommand(self, numSteps: int, check_detune: bool = True):
         # this is necessary because the tuners for the HLs move the other direction
         if self.cavity.cryomodule.is_harmonic_linearizer:
             numSteps *= -1
```

Comment:
Maybe this is the answer to my q above. You check here now?

Reply:
No, this is numSteps, which is slightly different (I should probably figure out a way to rewrite this lol). It's the number of steps requested by the user vs the maximum number of steps allowed by the stepper